### PR TITLE
Add plurality ballot support

### PR DESCRIPTION
**Pull request by ArendPeter**
⚠️NOTE: Highly recommend reviewing the individual commits instead of the code diff. They're much easier to follow

This commit does the following
 * Moves the StarBallotView into a GenericBallotView
 * Recreates StarBallotView to leverage GenericBallotView
 * Creates PluralityBallotView which also leverages GenericBallotView

I've got more more PRs coming for IRV and approval ;) 

image
image

### CODE REVIEW

File: frontend/src/components/Election/Voting/GenericBallotView.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+          </Grid>
+
+          <ColumnHeadings columns={columns}/>
+          <Divider style={{ width: '100%' }} />
+          <Rows candidates={candidates} scores={scores} onClick={onClick} columns={columns}/>
+
+          <Grid item xs={10} style={{ padding: '0.4cm 0cm' }}>
+            {race.num_winners == 1 && race.voting_method == 'STAR' &&
+              <Typography align='center' component="p">
+                This election uses STAR Voting and will elect 1 winner. In STAR Voting the two highest scoring candidates are finalists and the finalist preferred by more voters wins.
```
Comment (mikefranze) [inline frontend/src/components/Election/Voting/GenericBallotView.js:208]: We should move this to StarBallotView and add an input for the tabulation explanation to generic ballot view.

File: frontend/src/components/Election/Voting/GenericBallotView.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+    <Grid item xs={1}>
+      <Typography align='center' variant="h6" component="h6">
+        {columns.length == 1? ' ' : columnValue}
+      </Typography>
+    </Grid>
+  ));
+
+
+
+// Renders a complete RCV ballot for a single race
```
Comment (mikefranze) [inline frontend/src/components/Election/Voting/GenericBallotView.js:173]: nit: remove rcv references in comments in this file and in other ballot views. Not sure why it was there in the first place.

**Comment by mikefranze** [comment]
Looks good, though I'm wondering if we could reduce the whitespace on the right side.

File: frontend/src/components/Election/Voting/GenericBallotView.js (hunk 2 above)
Comment (ArendPeter) [reply frontend/src/components/Election/Voting/GenericBallotView.js:173]: Sounds good, will address this in the next PR as well

File: frontend/src/components/Election/Voting/GenericBallotView.js (hunk 1 above)
Comment (ArendPeter) [reply frontend/src/components/Election/Voting/GenericBallotView.js:208]: Sounds good, I've got a fix for this ready in the next PR

**Comment by ArendPeter** [comment]
> Looks good, though I'm wondering if we could reduce the whitespace on the right side.

Makes sense, I looked into this a little bit but I was having trouble figuring out which part of the styling to edit. When I inspected the element the color coding didn't match up with margin or padding?

image

**Comment by mikefranze** [comment]
How does that padding look with the expanded ballot button? You can enable that by adding a bio to a candidate. Another way you could do it is setting xs={4} or some other number. That controls how many columns in the grid the component will take up. For the star ballot the expand button is 1 column, the name is 4, and each of the six circles is 1.

**Comment by ArendPeter** [comment]
image
I just pushed a change making the text alignment depend on the ballot type, I think that'll be good for now?

I was struggling with the grid, my second preferred approach is to play more the padding, but that'll give problems for longer candidate names

**Review by mikefranze — APPROVED** [review]

**Pull request merged**